packages/web-components/src/utils/element-internals.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Check if the browser supports Custom States.
+ * @public
+ */
+export const CustomStatesSetSupported = CSS.supports('selector(:state(g))');
+
+/**
+ * This function is used to toggle a state on the control. If the browser supports Custom States, the state is toggled
+ * on the `ElementInternals.states` set. If the browser does not support Custom States, the state is toggled on the host
+ * element as an attribute with the format `state--{state}`.
```

Comment:
Seems good, I was confused by the wording because the https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals/states and https://developer.mozilla.org/en-US/docs/Web/API/CustomStateSet pages appears to show this is widely supported. Can you add a link to https://developer.mozilla.org/en-US/docs/Web/CSS/:state (added a sugggestion comment below).